packages/contracts/contracts/ChugSplashManager.sol

```diff
@@ -348,13 +383,44 @@ contract ChugSplashManager is Owned {
         // completed and reset the active bundle hash so that a new bundle can be executed.
         if (bundle.total == bundle.executions.length) {
             bundle.status = ChugSplashBundleStatus.COMPLETED;
+            bytes32 completedBundleId = activeBundleId;
             activeBundleId = bytes32(0);
 
-            emit ChugSplashBundleCompleted(activeBundleId, msg.sender, bundle.total);
+            emit ChugSplashBundleCompleted(completedBundleId, msg.sender, bundle.total);
             registry.announce("ChugSplashBundleCompleted");
+
+            // Refund the executor's bond
+            executorSelectionStrategy.returnExecutorBond(completedBundleId);
         }
```

Comment:
I left out the call that refunds the owner's bond after a bundle is completed. That way, owners can reuse bonds between bundles.

Reply:
Yep that's perfect. We can also easily represent this in our docs with something like "Because users could grief executors by cancelling upgrades shortly after starting them, we require that you have a minimum balance of X ETH in your Manager contract before you can start an upgrade. This balance acts as a deposit that you will forfeit if you need to cancel an active upgrade."